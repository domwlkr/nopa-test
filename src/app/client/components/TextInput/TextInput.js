import React from 'react';

const TextInput = (props) => {
  return (
    <div className="form-input">
      <label
        htmlFor={props.name}>
        {props.label}
      </label>
      <input
        id={props.name}
        name={props.name}
        type="text"
        placeholder={props.label}
        onChange={props.handleChange}
        data-error={props.errors} />
      <div className="errors">{props.errors}</div>
    </div>
  );
};

TextInput.propTypes = {
  name: React.PropTypes.string.isRequired,
  label: React.PropTypes.string.isRequired,
  handleChange: React.PropTypes.func,
  errors: React.PropTypes.string
};

export default TextInput;
