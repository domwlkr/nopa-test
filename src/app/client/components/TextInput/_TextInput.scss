.form-input {
  margin: 20px auto 50px;
  max-width: 330px;
  position: relative;

  label {
    display: none;
  }

  input {
    padding: 10px;
    width: 100%;
    max-width: 330px;
    border: 0;
    border-radius:1px;
    border-bottom: 3px solid lightgray;
  }

  .errors {
    position: absolute;
    width: 100%;
    left: 50%;
    top: calc(100% + 10px);
    transform: translateX(-50%);
  }
}
