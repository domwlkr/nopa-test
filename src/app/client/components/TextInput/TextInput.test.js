import React from 'react';
import TextInput from './TextInput';
import renderer from 'react-test-renderer';

const label = 'Surname';
const name = 'surname';
const handleChange = jest.fn();

test('Renders correctly', () => {
  const textInput = renderer.create(
    <TextInput name={name} label={label} handleChange={handleChange} />
  ).toJSON();

  expect(textInput).toMatchSnapshot();
});