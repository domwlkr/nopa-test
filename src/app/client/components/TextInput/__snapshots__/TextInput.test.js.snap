// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="form-input"
>
  <label
    htmlFor="surname"
  >
    Surname
  </label>
  <input
    data-error={undefined}
    id="surname"
    name="surname"
    onChange={[Function]}
    placeholder="Surname"
    type="text"
  />
  <div
    className="errors"
  />
</div>
`;
